Fails to build with default features off

**Describe the bug**

```
juice/juice$ cargo build --release --examples --no-default-features --features=cuda
...
   Compiling coaster-nn v0.5.0 (/mnt/bkel/vi/src/juice/coaster-nn)
error: external crate `rand_chacha` unused in `coaster_nn`: remove the dependency or add `use rand_chacha as _;`
    |
note: the lint level is defined here
   --> coaster-nn/src/lib.rs:107:5
    |
107 |     unused_crate_dependencies,
    |     ^^^^^^^^^^^^^^^^^^^^^^^^^
    = help: remove unnecessary dependency `rand_chacha`

error: could not compile `coaster-nn` due to previous error
```

Removing the denied lint leads to other error:

```
   Compiling juice v0.3.0 (/mnt/bkel/vi/src/juice/juice)
error[E0412]: cannot find type `Native` in this scope
   --> juice/src/layer.rs:786:40
    |
786 |         let native_backend = Backend::<Native>::default().unwrap();
    |                                        ^^^^^^ not found in this scope
```

Building with default features (i.e. `native`) enabled leads to benchmarks example being nonfunctional:

```
$ ../target/release/examples/benchmarks alexnet
Executing Model: "alexnet"
Examples run only with CUDA support at the moment, because of missing native convolution implementation for the Coaster NN Plugin.
Try running with `cargo run --release --no-default-features --features cuda --example benchmarks alexnet`.
```


**Please complete the following information:**

 - System: Debian bullseye
 - Version: 37911bed0353662cb4525599428c8bc0b0c262e0
 - Rust: `1.60.0-nightly (0c292c966 2022-02-08)`
 - Environment: cudnn version 8.3.2.44-1+cuda11.5 (no envvars, installed to `/usr/lib`).
 - Backends (if relevant):
   * opencl: -
   * cuda: NVIDIA-SMI 460.91.03    Driver Version: 460.91.03    CUDA Version: 11.2

--- a/coaster-nn/Cargo.toml
+++ b/coaster-nn/Cargo.toml
@@ -32,6 +32,6 @@ num = "0.4"
 [features]
 default = ["native", "cuda"]
 native = ["coaster/native", "rand", "rand_chacha"]
-cuda = ["coaster/cuda", "rcudnn", "libc", "thiserror", "rand_chacha"]
+cuda = ["coaster/cuda", "rcudnn", "libc", "thiserror"]
 opencl = ["coaster/opencl"]
 unstable = []

--- a/juice/examples/benchmarks.rs
+++ b/juice/examples/benchmarks.rs
@@ -9,6 +9,7 @@ use crate::co::prelude::*;
 use std::env;
 
 use std::rc::Rc;
+use std::sync::{RwLock, Arc};
 
 fn main() {
     env_logger::init();

